source/config/profileUpgrader.py

```diff
@@ -12,19 +12,22 @@
 
 SCHEMA_VERSION_KEY = "schemaVersion"
 
-def upgrade(profile, validator):
+def upgrade(profile, validator, writeProfileToFileFunc, shouldWriteProfileToFile):

```

Comment:
Why not just support profileFileFunc being None and therefore saving won't happen, rather than requring the extra boolean?